packages/webpack-config-flag-plugin/index.js

```diff
@@ -0,0 +1,100 @@
+const BasicEvaluatedExpression = require( 'webpack/lib/BasicEvaluatedExpression' );
+
+// Check that the given call expression is `config.isEnabled( 'flag' )` with
+// `config` as the default export or namespace, and return the `flag` literal value.
+const isCallOnDefaultOrNamespace = ( parser, moduleName, expr ) => {
+	return moduleName &&
+		parser.evaluate( moduleName ).identifier === 'imported var' &&
+		expr.callee.type === 'MemberExpression' &&
+		expr.callee.object.type === 'Identifier' &&
+		expr.callee.object.name === moduleName &&
+		expr.callee.property.type === 'Identifier' &&
+		expr.callee.property.name === 'isEnabled' &&
+		expr.arguments.length === 1 &&
+		expr.arguments[ 0 ].type === 'Literal'
+		? expr.arguments[ 0 ].value
+		: null;
+};
+
+// Check that the given call expression is `isEnabled( 'flag' )`
+// and return the `flag` literal value.
+const isNamedCall = ( parser, methodName, expr ) => {
+	return methodName &&
+		parser.evaluate( methodName ).identifier === 'imported var' &&
+		expr.callee.type === 'Identifier' &&
+		expr.callee.name === methodName &&
+		expr.arguments.length === 1 &&
+		expr.arguments[ 0 ].type === 'Literal'
+		? expr.arguments[ 0 ].value
+		: null;
+};
+
+const moduleTypes = [ 'javascript/auto', 'javascript/dynamic', 'javascript/esm' ];
+
+class ConfigFlagPlugin {
+	constructor( options ) {
+		this.flags = options && options.flags;
+	}
+
+	apply( compiler ) {
+		const handleParser = parser => {
+			// Hook into file changes.
+			parser.hooks.program.tap( 'ConfigFlagPlugin', () => {
+				// Reset imports on new file.
+				this.moduleName = undefined;
+				this.methodName = undefined;
+			} );
+
+			// Hook into imports.
+			parser.hooks.import.tap( 'ConfigFlagPlugin', ( statement, source ) => {
+				if ( source === 'config' ) {
+					// We have an import of 'config'.
+					const specifiers = statement.specifiers || [];
+
+					for ( const sp of specifiers ) {
+						// Default import (`import config from 'config'`)
+						if ( sp.type === 'ImportDefaultSpecifier' ) {
+							this.moduleName = sp.local && sp.local.name;
```

Comment:
The `sp.local` object is always present, isn't it? The specifier always refers some local binding, otherwise it wouldn't be there.

Reply:
Possibly? I tend to code rather defensively when dealing with undocumented APIs... Removed.

Reply:
The AST API is standardized and documented in the ESTree spec. `ModuleSpecifier` and its `local` field is documented here: https://github.com/estree/estree/blob/master/es2015.md#modulespecifier